### PR TITLE
Use dependency injection to allow alternative json parser or encoder

### PR DESCRIPTION
**Pull request by chiuczek**
Allow a custom/alternative JSON encode/decode function to be specified when sending a response or parsing a request. The `response.json` function gets an extra optional keyword argument `dumps`, with obvious default. When parsing a request, instead of the property `request.json` a method `request.load_json(loads=<new loads>)` can be used.

My use case is returning results from calculations, stored as numpy arrays or pandas dataframes. The pandas library has a nice JSON encode/decode module (derived from ujson) which will sensibly (and quickly) handle numpy arrays.

```{python}
import numpy as np
import pandas as pd
import sanic

def a_response_handler(request)
    request_data = request.load_json(loads=pd.json.loads)

# processing would go here

    response_data = {
        "first": np.array([0.0, 1.0, 4.0]),
        "second": np.random.uniform(0, 1, 10)
    }
    return sanic.response.json(response_data, dumps=pd.json.dumps)
```

### CODE REVIEW

**Comment by chiuczek** [comment]
This relates to issue #905, but wasn't inspired by it

**Comment by chiuczek** [comment]
Also, sorry for the username confusion, chiuczek is my personal account and intelliHugh is for work use, but they're both me!

File: sanic/request.py [hunk 1]
```diff
@@ -80,6 +80,16 @@ def json(self):
 
         return self.parsed_json
 
+    def load_json(self, loads=json_loads):
```
Comment (r0fls) [inline sanic/request.py]: you should replace https://github.com/chiuczek/sanic/blob/a8f764c1612ab17eb99924ac2363c05110a1911d/sanic/request.py#L74-L81 with a call to this function

Comment (chiuczek) [reply sanic/request.py]: of course I should, great suggestion, thanks

**Pull request merged**